pkg/reconciler/labeler/labels.go

```diff
@@ -45,28 +45,40 @@ func (c *Reconciler) syncLabels(ctx context.Context, r *v1.Route) error {
 	revisions := sets.NewString()
 	configs := sets.NewString()
 
-	// Walk the revisions in Route's .status.traffic and build a list
-	// of Configurations to label from their OwnerReferences.
-	for _, tt := range r.Status.Traffic {
-		rev, err := c.revisionLister.Revisions(r.Namespace).Get(tt.RevisionName)
-		if err != nil {
-			return err
+	// Walk the Route's .status.traffic and .spec.traffic and build a list
+	// of revisions and configurations to label
+	for _, tt := range append(r.Status.Traffic, r.Spec.Traffic...) {
+		revName := tt.RevisionName
+		configName := tt.ConfigurationName
+
+		latest := tt.LatestRevision != nil && *tt.LatestRevision
```

Comment:
I'd keep this in the if block below where it was in the first place. I went on looking if this is used somewhere else too.